functions/all_words.py

```
@@ -839,3 +839,132 @@ def dfs(state, req, failed, budget):
+    step_budget -- per-word DFS step cap; guards against a pathologically
+                   dense board blowing up the path search
+
+    Returns (results, total_playable). Each result is a dict:
+      {'word', 'len', 'pop', 'path': [node ids], 'nodes': [node ids]}
+    where 'path' is one concrete route through the board (for drawing the
+    word) and 'nodes' is the union of every tile the word can route through
+    (for the tap-a-tile-to-see-its-words view). Results are sorted longest
+    first, then most common, then alphabetically.
+    """
+    disabled = {int(n) for n in disabled} if disabled else set()
+    node_letter = {int(i): str(l).lower() for i, l in nodes.items()
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)